pkg/operator/starter.go

```diff
@@ -116,3 +123,35 @@ func replaceNamespaceFunc(namespace string) resourceapply.AssetFunc {
 		return bytes.ReplaceAll(content, []byte(namespaceKey), []byte(namespace)), nil
 	}
 }
+
+// getOperatorSyncState returns the management state of the operator to determine
+// how to sync conditional resources. It returns one of the following states:
+//
+//	Managed: resources should be synced
+//	Unmanaged: resources should NOT be synced
+//	Removed: resources should be deleted
+//
+// Errors fetching the operator state will log an error and return Unmanaged
+// to avoid syncing resources when the actual state is unknown.
+func getOperatorSyncState(operatorClient v1helpers.OperatorClientWithFinalizers) opv1.ManagementState {
```

Comment:
Few unit tests would be nice for this function

Reply:
Done